config/qthread_check_swapcontext.m4

```diff
@@ -65,9 +65,9 @@ AC_CACHE_CHECK([whether swapcontext works properly],
   [qthread_cv_swapcontext_works],
   [
   case "$host" in
-  *-darwin*)
-  qthread_cv_swapcontext_works=no
-  ;;
+#  *-darwin*)
```

Comment:
Since current x86 and M1 Macs seemingly support system ucontext, I think it is ok to drop this case statement.

Reply:
Consider adding something like 
 ```AS_IF([test "$host" in *-darvin*], [qt_host_based_enable_fastcontext=no],
    [qt_host_based_enable_fastcontext=yes]
	;;```
to L37 to disable fast context automatically on M1 Macs while the support is not available yet.

Reply:
darvin -> darwin.  But really, it's the combo of darwin and aarch64 (however that parses).  We don't want to disable fast context on earlier macs.

Reply:
Yes, that combo should come together in L37 if I see this correctly.

Reply:
https://github.com/janciesko/qthreads/tree/mac-allow-system-ucontext-2

Reply:
> https://github.com/janciesko/qthreads/tree/mac-allow-system-ucontext-2

This is not correct.  m1 macs are arm v8 not arm v7.  The arch to AC will be something like arm64 or aarch64.

Reply:
Yes, this was more an idea / place holder. We'd need to add the right case depending on what config gives us.

Reply:
We can match on `aarch64-*-darwin* | arm64*-*-darwin*`

Reply:
Actually, since it shouldn't really match any of the architectures in the case, it should fall through without any change.  I just tried from the qthreads repo branch (without the m4 change and without the disable fast context switch on configure) and it worked just fine.